google/cloud/storage/_opentelemetry_tracing.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Manages OpenTelemetry tracing span creation and handling. This is a PREVIEW FEATURE: Coverage and functionality may change."""
+
+import logging
+import os
+
+from contextlib import contextmanager
+
+from google.api_core import exceptions as api_exceptions
+from google.api_core import retry as api_retry
+from google.cloud.storage import __version__
+from google.cloud.storage.retry import ConditionalRetryPolicy
+
+
+ENABLE_OTEL_TRACES_ENV_VAR = "ENABLE_GCS_PYTHON_CLIENT_OTEL_TRACES"
+_DEFAULT_ENABLE_OTEL_TRACES_VALUE = False
+
+enable_otel_traces = os.environ.get(
+    ENABLE_OTEL_TRACES_ENV_VAR, _DEFAULT_ENABLE_OTEL_TRACES_VALUE
+)
+logger = logging.getLogger(__name__)
+
+try:
+    from opentelemetry import trace
+
+    HAS_OPENTELEMETRY = True
+
+except ImportError:
+    logger.debug(
+        "This service is instrumented using OpenTelemetry. "
+        "OpenTelemetry or one of its components could not be imported; "
+        "please add compatible versions of opentelemetry-api and "
+        "opentelemetry-instrumentation packages in order to get Storage "
+        "Tracing data."
+    )
+    HAS_OPENTELEMETRY = False
+
+_default_attributes = {
+    "rpc.service": "CloudStorage",
+    "rpc.system": "http",
+    "user_agent.original": f"gcloud-python/{__version__}",
+}
+
+
+@contextmanager
+def create_trace_span(
+    name, attributes=None, client=None, api_request=None, retry=None, **kwargs
```

Comment:
kwargs doesn't seem to be used here, what's its purpose?

Reply:
Removed for now; Will need it when we inject more span attributes in phase 2